### ISSUE TITLE
Lab1 - word2vec train loss

### ISSUE THREAD

**Issue by angeloskanatas** [slp-ntua/slp-labs#104]
Καλησπέρα, όταν εκπαιδεύουμε το word2vec, το loss σε κάθε epoch είναι 0 (χρησιμοποιούμε το δοθέν W2VLossLogger()). Αυτό γιατί μπορεί να συμβαίνει;

**Comment by efthymisgeo** [comment]
Στείλε μου τι εντολή ετρεξες

**Comment by angeloskanatas** [comment]
Κάνω train το μοντέλο με:  `model.train(sentences, total_examples=len(sentences), epochs=epochs, callbacks=[W2VLossLogger()])`, όπου η κλάση `W2VLossLogger` log-άρει το loss μετά από κάθε epoch με αυτόν τον τρόπο:
```
class W2VLossLogger(CallbackAny2Vec):
  """
  Callback to print loss after each epoch.
  Use by passing model.train(..., callbacks=[W2VLossLogger()])
  """

  def __init__(self):
    self.epoch = 0

  def on_epoch_end(self, model):
    loss = model.get_latest_training_loss()

    if self.epoch == 0:
      print("Loss after epoch {}: {}".format(self.epoch, loss))
    else:
      print(
        "Loss after epoch {}: {}".format(
          self.epoch, loss - self.loss_previous_step
          )
      )
    self.epoch += 1
    self.loss_previous_step = loss

```

**Comment by efthymisgeo** [comment]
Που ορίζεις το μοντέλο;

**Comment by angeloskanatas** [comment]
Έχω συμπληρώσει την συνάρτηση `train_w2v_model` που έχει δοθεί. Αρχικοποιώ με: `model = Word2Vec(sentences, vector_size=embedding_dim, window=window, min_count=min_word_count, workers=workers-1, sg=1, compute_loss=True) # skip-gram model`.

**Comment by IagoDesperado** [comment]
Εχω και εγώ το ίδιο πρόβλημα με τον συνάδελφο.  Κάνω train ως εξής 
model.train(
        sentences=sentences,
        total_examples=len(sentences),
        corpus_iterable=sentences,
        epochs=epochs,
        callbacks=[W2VLossLogger()],

    )
Αρχικοποιώ το μοντέλο ως εξής;

    model = Word2Vec(
        sentences=sentences,
        window=window,
        workers=workers,
        min_count=min_word_count,

    )

    # Build model

  
 Αν χρειάζεται να στείλω κάτι άλλο σε κώδικα, μπορείτε να μου πείτε

**Comment by Con-Hal** [comment]
Εχω και εγώ το ίδιο πρόβλημα με τον συνάδελφο. Κάνω train ως εξής
model.train(
sentences=sentences,
total_examples=len(sentences),
corpus_iterable=sentences,
epochs=epochs,
callbacks=[W2VLossLogger()],

)

Αρχικοποιώ το μοντέλο ως εξής;

model = Word2Vec(
    sentences=sentences,
    window=window,
    workers=workers,
    min_count=min_word_count,

)

# Build model

Αν χρειάζεται να στείλω κάτι άλλο σε κώδικα, μπορείτε να μου πείτε

**Comment by angeloskanatas** [comment]
Για όσους δεν έχουν βρει την λύση σε αυτό απλά κάντε: `model.train(sentences, total_examples=len(sentences), epochs=epochs, compute_loss=True, callbacks=[W2VLossLogger()])` (δεν χρειάζεται να προσθέσετε τις παραμέτρους `compute_loss` και `callbacks` στην αρχικοποίηση του μοντέλου).

**Comment by Con-Hal** [comment]
Το να προσθέσω πράγματι compute_loss=True στην model.train το έλυσε για εμένα.
